crates/stackable-operator/src/builder/pod/container.rs

```diff
@@ -21,11 +22,25 @@ pub enum Error {
         source: validation::Errors,
         container_name: String,
     },
+
+    #[snafu(display(
+        "The volumeMount is clashing with an already existing volumeMount with the same mountPath but a different content. \
+            The existing volumeMount is {existing_volume_mount:?}, the new one is {new_volume_mount:?}"
+    ))]
+    ClashingVolumeMountMountPath {
+        existing_volume_mount: Box<VolumeMount>,
+        new_volume_mount: Box<VolumeMount>,
+    },
 }
 
 /// A builder to build [`Container`] objects.
 ///
 /// This will automatically create the necessary volumes and mounts for each `ConfigMap` which is added.
+///
+/// This struct is often times using an [`IndexMap`] to have consistent ordering (so we don't produce reconcile loops).
+/// We are also choosing it over an [`std::collections::BTreeMap`], as it's easier to debug for users, as logically
+/// grouped volumeMounts (e.g. all volumeMounts related to S3) are near each other in the list instead of "just" being
```

Comment:
Didn't a plain Vec fulfill all these criteria already ?

Reply:
Yes, but the problem with a Vec is that it contain duplicates (multiple volumes with the same name), which k8s will reject

Reply:
I understand that this is what this PR is fixing.
The question was, what problem does IndexMap solve that the plain Vec and a filter couldn't ?

Reply:
Ah ok, now I understand.

For me it feels "right" to use a data-structure modeling your data. This way it's 100% guaranteed that there are no duplicates. If we leave it as Vec, chances are, in the future someone might just `.push()` something onto it without knowing about our current problems, thus introducing the bug again. This can not happen with a Map (regardless if BTreeMap or IndexMap). I e.g. ran into this as the list comming in from `add_volumes` itself can already contain duplicates, so we can not simply `extend` that on to the Vec. 

Additionally it is more performant, as the Vec would have an o(n) insert cost, while the IndexMap has o(1) - but I would consider this less important